app/src/androidTest/kotlin/com/google/samples/apps/nowinandroid/ui/FakeWindowMetricsCalculatorRule.kt

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2024 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.samples.apps.nowinandroid.ui
+
+import android.app.Activity
+import android.content.Context
+import android.graphics.Rect
+import androidx.compose.ui.geometry.Size
+import androidx.compose.ui.unit.toIntSize
+import androidx.window.layout.WindowMetrics
+import androidx.window.layout.WindowMetricsCalculator
+import androidx.window.layout.WindowMetricsCalculatorDecorator
+import org.junit.rules.TestRule
+import org.junit.runner.Description
+import org.junit.runners.model.Statement
+
+/**
+ * A test rule which allows overriding the reported WindowMetrics with an arbitrary size, defaulting
+ * to [Size.Zero]. The size is not persisted and will be reset after each test.
+ */
+class FakeWindowMetricsCalculatorRule : TestRule {
+
+    private val calculator = FakeWindowMetricsCalculator()
+
+    fun setWindowSize(size: Size) {
+        calculator.windowSize = size
+    }
+
+    override fun apply(base: Statement?, description: Description?): Statement {
+        return object : Statement() {
+            override fun evaluate() {
+                WindowMetricsCalculator.overrideDecorator(
```

Comment:
(commenting for visibility)

This is a `@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)` method, so while it may work to call, it's not an API explicitly supported by the library for us to use until that is removed in the library.